### PR TITLE
fix: print help text to stderr

### PR DESCRIPTION
**Pull request by moolen**
The help text should be printed to stderr
to avoid confusion when piping the stdout to another program.
e.g. `cilium endpoint confi 173  | grep Conntrack`

Fixes: #2498

```release-note
cmd: help text prints to stderr
```

I'm not sure if `tests/*.sh` is the right place for testing stdout/err behavior

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/6709)

### CODE REVIEW

**Comment by joestringer** [comment]
test-me-please

**Comment by aanm** [comment]
@moolen the `tests/` directory is deprecated, we now use `test` with ginkgo. More info: http://docs.cilium.io/en/v1.3/contributing/#end-to-end-testing-framework

**Review by joestringer — APPROVED** [review]
Hi @moolen , thanks for the contribution!

I tested this out locally and saw that `cilium version` output correctly goes to stdout, while the `cilium help` output goes to stderr. This matches the description in issue #2498.

For future reference, the bash-based under `tests/` is largely deprecated and is no longer triggered from the automatic CI. That said, it's good to see that the functionality was validated.

PR LGTM

**Review by aanm — APPROVED** [review]

**Comment by moolen** [comment]
Thanks @aanm and @joestringer! I'll look into implementing the test w/ ginkgo

File: test/runtime/cli.go [hunk 1]
```diff
@@ -112,4 +112,26 @@ var _ = Describe("RuntimeCLI", func() {
 		})
 	})
 
+	Context("stdout stderr testing", func() {
+
+		It("root command help should print to stderr", func() {
```
Comment (joestringer) [inline test/runtime/cli.go:117]: As a matter of style, ginkgo encourages using `It()` statements in the way you would with english sentences, ie

`It("tests that the root command help output prints to stderr")`

**Review by joestringer — APPROVED** [review]
New tests look good, one minor note for style if you happen to define additional tests in future.

**Comment by joestringer** [comment]
test-me-please

**Comment by joestringer** [comment]
Depending on which master commit this branch is based upon, it may hit some of the master Ginkgo failures seen recently but the main thing I'm looking for in the ginkgo run is to validate that the new test runs and succeeds in the CI environment.

**Pull request merged**